### PR TITLE
T22873 Various fixes to unit tests

### PR DESCRIPTION
**Pull request by pwithnall**
https://phabricator.endlessm.com/T22873

### CODE REVIEW

File: test-common/utils.c [hunk 1]
```diff
@@ -1643,6 +1644,7 @@ spawn_updater (GFile *sysroot,
       { "OSTREE_REPO", NULL, repo },
       { "OSTREE_SYSROOT_DEBUG", "mutable-deployments", NULL },
       { "EOS_DISABLE_METRICS", "1", NULL },
+      { "G_DEBUG", fatal_warnings ? "gc-friendly,fatal-warnings" : "gc-friendly", NULL },
```
Comment (wjt) [inline test-common/utils.c]: `"fatal-warnings,gc-friendly" + (15 * !!fatal_warnings)`? ;-)

File: test-common/flatpak-spawn.c [hunk 2]
```diff
@@ -182,6 +183,7 @@ flatpak_build_export (GFile        *updater_dir,
       { NULL, branch },
       { "gpg-sign", keyid },
       { "gpg-homedir", gpg_homedir_path },
+      { (collection_id) != NULL ? "collection-id" : NULL, collection_id },
```
Comment (wjt) [inline test-common/flatpak-spawn.c]: Why the parens here?

File: test-common/utils.c (hunk 1 above)
Comment (pwithnall) [reply test-common/utils.c]: Ouch, no!

File: test-common/flatpak-spawn.c (hunk 2 above)
Comment (pwithnall) [reply test-common/flatpak-spawn.c]: I can not computer.

Comment (pwithnall) [reply test-common/flatpak-spawn.c]: Fixup pushed.

**Comment by pwithnall** [comment]
Jenkins failure:
```
# eos-updater-tests-DEBUG: Trying to cancel state Ready
# eos-updater-tests-DEBUG: Error cancelling Ready: GDBus.Error:com.endlessm.Updater.Error.WrongState: Can't call Cancel() while in state Ready (nothing to be cancelled)
# eos-updater-tests-DEBUG: State changed Ready
# eos-updater-tests-DEBUG: Trying to cancel state Polling
# eos-updater-tests-DEBUG: Cancelled state Polling
# eos-updater-tests-DEBUG: Trying to cancel state Error
# eos-updater-tests-DEBUG: Error cancelling Error: GDBus.Error:com.endlessm.Updater.Error.WrongState: Can't call Cancel() while in state Error (nothing to be cancelled)
# eos-updater-tests-DEBUG: State changed Error
# eos-updater-tests-DEBUG: Error name: com.endlessm.Updater.Error.Cancelled
# eos-updater-tests-DEBUG: Error message: Operation was cancelled
# eos-updater-tests-DEBUG: State changed Polling
# eos-updater-tests-DEBUG: Trying to cancel state None
# eos-updater-tests-DEBUG: Error cancelling None: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name com.endlessm.Updater was not provided by any .service files
# eos-updater-tests-DEBUG: State changed None
# eos-updater-tests:ERROR:../../../tests/test-update-direct.c:293:test_cancel_update: 'timed_out' should be FALSE
```

The `# eos-updater-tests-DEBUG: Error cancelling None: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name com.endlessm.Updater was not provided by any .service files` line is quite suspicious. I suspect it means that `eos-updater` crashed.

Probably because of the combination of `G_DEBUG=fatal-warnings` and this critical warning:
```
(/build/src/eos-updater_1.3.0~rc3+dev1051.2c0b4be/_build/sub/eos-updater/.libs/eos-updater:29780): EosMetrics-CRITICAL **: Unable to connect to the D-Bus event recorder server: Could not connect: No such file or directory
```

**Comment by pwithnall** [comment]
Pushed another commit which fixes the metrics problem. The test failure now is:
```
ERROR: test-update-broken-delta - too few tests run (expected 1, got 0)
```

That’s expected — it’s because the [OSTree changes](https://github.com/endlessm/ostree/pull/122) aren’t available in OBS yet.

**Comment by pwithnall** [comment]
So this can’t be merged until the OSTree changes are in. Also, it **needs autosquashing** before merging.

**Comment by pwithnall** [comment]
bot, retest this please

**Comment by pwithnall** [comment]
The tests passed, but there was a merge conflict due to recent changes on master. I’ve rebased (the conflict was trivial) and will wait for Jenkins to pass again before merging.

**Pull request merged**